### PR TITLE
Shape to mermaid

### PR DESCRIPTION
**Pull request by pheyvaer**

### CODE REVIEW

**Comment by pheyvaer** [comment]
@pietercolpaert Can you have a look at this PR? The code to generate the Mermaid representation is in `lib/ShapesGraph.ts`. You find the corresponding tests at `tests/07 - mermaid/mermaid.test.ts`.

**Comment by ajuvercr** [comment]
@pheyvaer Looking pretty cool!
Would it be possible to change 'shape' in the diagram to the last part of the named node of the shape id?
```turtle
ex:SequencePathShape a sh:NodeShape ;
    sh:property [
        sh:path (ex:p1 ex:p2 ) ;
        sh:minCount 1
    ] .
 ```

**Comment by pietercolpaert** [comment]
I like the idea, but do we keep that information in our structure? Not sure

**Comment by julianrojas87** [comment]
I made some changes locally, following @ajuvercr's suggestion and it can produce this:
image

If this is approved, I can push the changes (@pietercolpaert?)

**Comment by pietercolpaert** [comment]
Approved!

Op do 2 mei 2024 18:17 schreef Julian Rojas ***@***.***>:

File: lib/ShapesGraph.ts [hunk 1]
```diff
@@ -367,6 +367,16 @@ export class ShapesGraph {
     nodeShapeId: Term,
     shape: ShapeTemplate,
   ) {
+    // Extract label
+    const rdfsLabel = getObjects(shapeStore, nodeShapeId, RDFS.terms.label)[0];
```
Comment (pheyvaer) [inline lib/ShapesGraph.ts:371]: @julianrojas87 Doesn't this give an error when no objects is found?

Comment (pietercolpaert) [reply lib/ShapesGraph.ts:371]: Apparently it doesn’t

Comment (julianrojas87) [reply lib/ShapesGraph.ts:371]: `getObjects()` returns an empty array if no match is found. So I think the code is safe as it is.

**Review by pietercolpaert — APPROVED** [review]

**Pull request merged**